platform/vcs-impl/src/com/intellij/openapi/vcs/update/AbstractCommonUpdateAction.java

```diff
@@ -301,7 +301,7 @@ private static boolean supportingVcsesAreEmpty(final ProjectLevelVcsManager vcsM
     private LocalHistoryAction myLocalHistoryAction;
 
     public Updater(final Project project, final FilePath[] roots, final Map<AbstractVcs, Collection<FilePath>> vcsToVirtualFiles) {
-      super(project, getTemplatePresentation().getText(), true, VcsConfiguration.getInstance(project).getUpdateOption());
+      super(project, getTemplatePresentation().getText(), true, ActionInfoEx.maybeOverrideBackGroundOption(myActionInfo, VcsConfiguration.getInstance(project).getUpdateOption()));
```

Comment:
This does not apply to normal VCSes. Even in our VCS implementation most updates can happen in the background.  However our special case we have a concept of layers, similar to ClearCase.  Switching layers may affect the SDK,  console environment, state of several controls, etc.  We'd really rather prefer avoid handling complex in-between states for this operation right now and just perform it modally.